### PR TITLE
Add forms

### PR DESCRIPTION
**Pull request by Mahamadou-Nouridine**
In this Pr, I brought changes according to the following requirements:
-  Create a method called `current_user` in your ApplicationController to make current user data available to all controllers.
    -  It will return the first user from the database.
    -  Don't worry about proper authentication at this time. You will add it in the upcoming projects.
- Create forms to perform the following functions:
    - Creates a Post on behalf of the `current_user` (use the method that you created in your ApplicationController).
    - Create a comment on behalf of the `current_user` (use the method that you created in your ApplicationController).
- Allow Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -8,4 +8,15 @@ def show
...
+  def new
+    @user = User.find(params[:user_id])
+    render 'new'
+  end
+
+  def create
+    user = current_user
+    Post.create(title: params[:title], text: params[:text], author_id: user.id)
+    redirect_to "/users/#{user.id}/posts", notice: 'new post created successfully'
+  end
```
Comment (nahnahdev) [inline app/controllers/posts_controller.rb]: - [ ] Great job so far @Mahamadou-Nouridine There are just a few changes I will suggest you implement and you'll be on your way to the next milestoone :wink: 
- [ ] To begin, when I tried to create a new post, I was faced with an error and per the project requirements, you need to ensure that a post is created on behalf of the currently logged-in user (current_user) of the app hence making use of the `current_user` method you created in the `application_controller.rb`. :wink: 
- [ ] So kindly ensure that the `current_user` is able to create new posts successfully without any errors :100:
- Error Message 

[Screencast from 08-25-2023 12:54:07 PM.webm](https://github.com/Mahamadou-Nouridine/blog-app/assets/102699958/2f4fcf47-0f2c-4be0-b2f7-954973230117)

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -34,7 +34,7 @@
           <%end%>
           <%="#{@user.name} has no any post" unless !@user.posts.empty?%>
         </ul>
-        <%=button_to "See all posts", "/users/#{@user.id}/posts", class: 'all-posts', method: 'get'%>
+        <%=button_to "See all posts", "/users/#{@user.id}/posts", class: 'all-posts', method: :get%>
```
Comment (nahnahdev) [inline app/views/users/show.html.erb:37]: - [x] Currently, when the `See all posts ` button is clicked, it leads to an error page so kindly fix that as well :wink: 
- [x] You can consider handling the case where there are no posts, that way a user will be navigated to the form to add a new post, and if there are posts they are Navigated to the `posts index` page :100: 
- Error Message

[Screencast from 08-25-2023 12:57:18 PM.webm](https://github.com/Mahamadou-Nouridine/blog-app/assets/102699958/fb08840f-df36-48d9-b073-89d5cf3c4b48)

- [x] Once this is fixed, kindly ensure that your comment and like features work as expected so the next reviewer can check if they all work as expected :wink:

File: config/routes.rb [hunk 3]
```diff
@@ -3,7 +3,13 @@
 
   # Defines the root path route ("/")
   root "users#index"
-  get "/users/:id", to: "users#show"
-  get "/users/:user_id/posts", to: "posts#index"
-  get "/users/:user_id/posts/:id", to: "posts#show"
+  # get "/users/:id", to: "users#show"
+  # get "/users/:user_id/posts", to: "posts#index"
+  # get "/users/:user_id/posts/:id", to: "posts#show"
```
Comment (nahnahdev) [inline config/routes.rb]: - [x] Kindly avoid leaving commented pieces of code in your codebase. It makes your codebase unclean and is not a good practice :+1:

**Review by nahnahdev — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi @Mahamadou-Nouridine , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Highlights :fire: 
- Linters well setup and passing :heavy_check_mark: 
- Professional README :heavy_check_mark: 
- ApplicationController contains `#current_user` method :heavy_check_mark: 
- All required routes present in the `routes.rb` file :heavy_check_mark: 
- Form present in the `views/posts/new.html.erb` file which accepts input for creating posts :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (Mahamadou-Nouridine) [reply app/controllers/posts_controller.rb]: Thank you very much @NahnahAJ, I've looking for the reason why the post is not being created. Actually, this review submission is to get a clue. Thanks again

**Review by ichala — APPROVED** [review]
Hi Student  ,
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
 
68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966

# To Highlight 💪🏻

- Changes Implemented ✔️ 


 Good job implementing the requirements  :+1: Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Optional Suggestions  :

- N/A


**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag  ichala in your question so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**